### PR TITLE
feat(FEC-13507): support doc entries

### PR DESCRIPTION
**Pull request by semarche-kaltura**
base PR: https://github.com/kaltura/playkit-js-doc-player/pull/1

solves: https://kaltura.atlassian.net/browse/FEC-13507

Screenshot 2024-01-16 at 12 53 04

### CODE REVIEW

File: translations/en.i18n.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
       "quiz_type": "Quiz",
       "live_type": "Live",
       "image_type": "Image",
+      "document_type": "Document",
```
Comment (SivanA-Kaltura) [inline translations/en.i18n.json:16]: is there a i18n ticket ?

File: src/components/playlist-item/index.tsx [hunk 2]
```diff
@@ -93,6 +107,24 @@ export const PlaylistItem = withText(translates)(({item, active, onSelect, plugi
         </div>
       );
     }
+    if (type === MediaTypes.Document) {
```
Comment (SivanA-Kaltura) [inline src/components/playlist-item/index.tsx:110]: should this be in common ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: translations/en.i18n.json (hunk 1 above)
Comment (semarche-kaltura) [reply translations/en.i18n.json:16]: yes https://kaltura.atlassian.net/browse/FEC-13644

File: src/components/playlist-item/index.tsx (hunk 2 above)
Comment (semarche-kaltura) [reply src/components/playlist-item/index.tsx:110]: that requires refactoring of other playlist-item type icons, for now playlist icons keeps in local repo

File: src/components/playlist-item/index.tsx [hunk 3]
```diff
@@ -36,26 +37,38 @@ interface PlaylistItemProps {
...
     return Number(progress.toFixed());
   }, [sources, viewHistory]);
 
   const renderAddons = useMemo(() => {
-    if (sources.type === core.MediaType.LIVE) {
+    const type = sources.mediaEntryType || sources.type;
+    if ([MediaTypes.Image, MediaTypes.Document].includes(type)) {
+      return null;
+    }
+    if (sources.mediaEntryType === MediaTypes.Live) {
```
Comment (lianbenjamin) [inline src/components/playlist-item/index.tsx:71]: you can use `type` instead of `sources.mediaEntryType`

**Review by lianbenjamin — CHANGES_REQUESTED** [review]

File: src/components/playlist-item/index.tsx [hunk 4]
```diff
@@ -36,26 +37,38 @@ interface PlaylistItemProps {
...
+  document?: string;
   toPlayAreaLabel?: string;
   currentlyPlaying?: string;
   playlistItemIndex?: string;
 }
 
+const MediaTypes = {
+  Live: core.MediaType.LIVE,
+  Image: core.MediaType.IMAGE,
+  Document: 'Document' // TODO: import from core.MediaType.DOCUMENT
```
[the remark this answers is not in the document]
Comment (semarche-kaltura) [reply src/components/playlist-item/index.tsx:49]: once related PR got merged - I'll change

**Pull request merged**